s3repo/model.py

```diff
@@ -625,8 +619,24 @@ def get_directory(self, path):
         # for more detaied description.
         list_parameters = {'Bucket': self.bucket.name,
                            'Delimiter': '/',
+                           'MaxKeys': 1,
                            'Prefix': abs_path}
 
+        # Check the existence of the directory.
+        # https://stackoverflow.com/a/68910145
+        objects = self.s3_client.list_objects_v2(**list_parameters)
+        if not 'CommonPrefixes' in objects:
+            raise RuntimeError('No such directory.')
```

Comment:
Will it lead to 404 or 500 HTTP code? Should it be 404?

Reply:
In this place it will lead to a 500 error, which on the one hand leaves the previous logic in force, but on the other hand, it is actually a second check for the existence of the directory and if the user incorrectly specified the directory will return exactly 404 on the results of the previous check.